### ISSUE TITLE
Unused parameters in loss computation breaks DDP autograd trace

### ISSUE THREAD

**Issue by Ben-Schneider-code** [openai/Video-Pre-Training#53]
In the forward pass of `InverseActionNet` should the output of `self.lastlayer` be passed into `final_ln`? The current implementation breaks torch.DDP training as the parameters in the `self.lastlayer` module are not used in the output (loss computation) -- which crashes autograd.
1. Which implementation is the architecture suppposed to be?
2. What implementation are the provided checkpoints trained with?

```    
def forward(self, ob, state_in, context):
        first = context["first"]
        x = self.img_preprocess(ob["img"])

        # Conv3D Prior to Impala
        if self.conv3d_layer is not None:
            x = self._conv3d_forward(x)

        # Impala Stack
        x = self.img_process(x)

        if self.recurrent_layer is not None:
            x, state_out = self.recurrent_layer(x, first, state_in)

        x = F.relu(x, inplace=False)

        pi_latent = self.lastlayer(x) # pi_latent is immediately discarded?
        pi_latent = self.final_ln(x) # should be self.final_ln(pi_latent) or self.final_ln(pi_latent + x)?
        return (pi_latent, None), state_out
```

**Comment by Miffyli** [comment]
Hey! That's a good catch, which funnily has gone under the radar all this time 😆 .

I do not have the source to look at, but from my memory, all this code should be quite copy-paste from the original code, including the "bug". So with that:

1) This is the implementation it was trained on (but probably with some DDP or other logic that did not detect unused tensors)
2) The checkpoints are with this broken logic

Also from memory, I think many people used the IDM in Minecraft and even in other games succesfully to decode the right actions, so I believe the current implementation is correct. I would still maybe pass it through the dataset to see how often it outputs the right action (and maybe try "fixing" the bug, to see if performance gets better or worse).

**Comment by Ben-Schneider-code** [comment]
Okay, thank you for the reply!
